packages/core/src/tracing/instrumentation/messaging/rdkafka.js

```diff
@@ -44,17 +44,23 @@ exports.activate = function activate(extraConfig) {
     }
     extraConfigHeader = extraConfig.tracing.kafka.headerFormat;
   }
-  logWarningForKafkaHeaderFormat(extraConfigHeader || configHeader);
+  if (extraConfigHeader || configHeader) {
+    logWarningForKafkaHeaderFormat(extraConfigHeader || configHeader);
+  }
   isActive = true;
 };
 
 exports.deactivate = function deactivate() {
   isActive = false;
 };
 
-// Note: This function can be removed as soon as we finish the Kafka header migration and remove the ability to
-// configure the header format (at that point, we will only be using string headers).
+// Note: This function can be removed as soon as we finish the Kafka header migration phase 2 and remove the ability to
+// configure the header format.
```

Comment:
```suggestion
// configure the header format. Might happen in major release v4.
```

Reply:
Background: I usually enter e.g. "v4" in the IDE search to look for any code which needs a cleanup.